service/matching/taskReader.go

```diff
@@ -134,28 +131,27 @@ dispatchLoop:
 			if !ok { // Task list getTasks pump is shutdown
 				break dispatchLoop
 			}
-			isolationGroup := ""
-			if tr.isIsolationEnabled() && len(taskInfo.PartitionConfig) > 0 {
-				taskInfo.PartitionConfig[partition.WorkflowRunIDKey] = taskInfo.RunID
-				ctx, cancel := context.WithTimeout(tr.cancelCtx, time.Second)
-				group, err := tr.partitioner.GetIsolationGroupByDomainID(ctx, taskInfo.DomainID, taskInfo.PartitionConfig)
-				if err != nil {
-					tr.logger.Error("Failed to get isolation group for async match task", tag.TaskID(taskInfo.TaskID), tag.Error(err))
-				} else if group != "" {
-					isolationGroup = group
-				}
-				cancel()
-			}
-			task := newInternalTask(taskInfo, tr.completeTask, types.TaskSourceDbBacklog, "", false, nil, isolationGroup)
 			for {
```

Comment:
Question: I don't quite understand this, what's the second for-loop for?

Reply:
keep trying to dispatch a single task to pollers